src/govee_ble/parser.py

```diff
@@ -171,42 +171,61 @@ def _process_mfr_data(
             data = data[:-25]
             msg_length = len(data)
             if debug_logging:
-                _LOGGER.debug("Cleaned up packet: %s %s", mgr_id, hex(data))
-
-        if msg_length == 24 and (
-            (is_5121 := "GV5121" in local_name)
-            or (is_5122 := "GV5122" in local_name)
-            or (is_5123 := "GV5123" in local_name)
-            or (is_5125 := "GV5125" in local_name)
-            or (is_5126 := "GV5126" in local_name)
-        ):
-            sensor_type = SensorType.BUTTON
-            if is_5121:
-                self.set_device_type("H5121")
-                sensor_type = SensorType.MOTION
-            elif is_5122:
-                self.set_device_type("H5122")
-            elif is_5123:
-                self.set_device_type("H5123")
-                sensor_type = SensorType.WINDOW
-            elif is_5125:
-                self.set_device_type("H5125")
-            elif is_5126:
-                self.set_device_type("H5126")
-            b_front_of_device_id = data[:2]
-            assert b_front_of_device_id
+                _LOGGER.debug("Cleaned up packet: %s %s", mgr_id, data.hex())
+
+        if msg_length == 24:
+            front_of_device_id = data[:2]
+            assert front_of_device_id
             time_ms = data[2:6]
             enc_data = data[6:22]
             enc_crc = data[22:24]
             if not calculate_crc(enc_data) == int.from_bytes(enc_crc, "big"):
                 _LOGGER.warning("CRC check failed for H512x: %s", hex(data))
                 return
+
             key = time_ms + bytes(12)
             try:
                 decrypted = decrypt_data(key, enc_data)
             except ValueError:
                 _LOGGER.warning("Failed to decrypt H512x: %s", hex(data))
                 return
+            model_id = decrypted[2]
```

Comment:
@gh2o  `model_id` is a guess here. It seems to hold up as I bought multiple of the same device to confirm it, but it would be great if you had some insight here since I feel like I'm poking around in the dark with this one.

Reply:
I only have the H5122 (single button), and the FW appears to hard-code `decrypted[0/1/2/3]` so I'm not entirely sure what the meaning of these bytes are.

One possible hint though is that the FW code does different operations to assign to those bytes. To summarize, it does something like this:

```python
decrypted[0] = ord('1') - 0x30
decrypted[1] = ord('5') - 0x30
decrypted[2] = 8
decrypted[3] = 2
```

If I had to venture a wild guess, the meaning of the numbers would be:
- `decrypted[0]` = version (major)
- `decrypted[1]` = version (minor)
- `decrypted[2]` = model ID
- `decrypted[3]` = number of data bytes following this byte

Reply:
Would it make sense to do

> "GV5121" in local_name **and** model_id == 3

instead of **or** so that if this invariant breaks in the future, we can catch it?

Reply:
The local name is empty when the scanner is passive

Reply:
I bought another motion sensor and 2 button remote.  This pattern held true so I think we are ok here at least for now

Reply:
Great, good to know.